### PR TITLE
Upgrade Azure SDK to v12

### PR DESCRIPTION
**Pull request by EslaMx7**

### CODE REVIEW

File: ExampleProject/ExampleProject.csproj [hunk 1]
```diff
@@ -13,18 +13,18 @@
   <ItemGroup>
     <PackageReference Include="MessagePack" Version="2.1.165" />
     <PackageReference Include="MessagePack.ImmutableCollection" Version="2.1.165" />
-    <PackageReference Include="Microsoft.Azure.Storage.Blob" Version="9.4.1" />
+	<PackageReference Include="Azure.Storage.Blobs" Version="12.12.0" />
     <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
   </ItemGroup>
   <!-- When developing locally, comment out this block and uncomment the one below it. -->
-  <ItemGroup>
+  <!--<ItemGroup>
```
Comment (VictorNicollet) [inline ExampleProject/ExampleProject.csproj]: The comments should be restored to their original state.

File: Lokad.AzureEventStore.Cache/AzureCacheProvider.cs [hunk 2]
```diff
@@ -51,36 +54,38 @@ public AzureCacheProvider(CloudBlobContainer container)
...
-
-                ct = list.ContinuationToken;
-
-                foreach (var item in list.Results)
-                    if (item is CloudBlockBlob blob)
+                var list = _container.GetBlobsAsync(traits: BlobTraits.Metadata,
+                   states: BlobStates.None,
+                   prefix: stateName,
+                   cancellationToken: default(CancellationToken))
+                   .AsPages(token);
```
Comment (VictorNicollet) [inline Lokad.AzureEventStore.Cache/AzureCacheProvider.cs]: My understanding is that iterating over `.AsPages(null)` will result in all blobs being enumerated, so there is no need to keep track of the continuation token, or have an outer `while` loop.

File: Lokad.AzureEventStore.Cache/Lokad.AzureEventStore.Cache.csproj [hunk 3]
```diff
@@ -14,19 +14,19 @@
 
   <!-- When working locally, comment out this block and uncomment the one below. -->
 
-  <ItemGroup>
+  <!--<ItemGroup>
```
Comment (VictorNicollet) [inline Lokad.AzureEventStore.Cache/Lokad.AzureEventStore.Cache.csproj]: Comments  need to be restored.

File: Lokad.AzureEventStore.Test/Lokad.AzureEventStore.Test.csproj [hunk 4]
```diff
@@ -24,17 +24,14 @@
   </ItemGroup>
   
   <!-- When developing locally, comment out this block and uncomment the one below it. -->
-  <ItemGroup>
+  <!--<ItemGroup>
```
Comment (VictorNicollet) [inline Lokad.AzureEventStore.Test/Lokad.AzureEventStore.Test.csproj]: Comments need to be restored.

File: Lokad.AzureEventStore/Drivers/AzureHelpers.cs [hunk 5]
```diff
@@ -113,28 +117,33 @@ public static int ParseNth(string str)
...
+                    states: BlobStates.None,
                     prefix: Prefix,
-                    blobListingDetails: BlobListingDetails.Metadata,
-                    maxResults: null,
-                    currentToken: token,
-                    options: new BlobRequestOptions(),
-                    operationContext: new OperationContext(),
-                    cancellationToken: cancel);
+                    cancellationToken: cancel)
+                    .AsPages(token);
```
Comment (VictorNicollet) [inline Lokad.AzureEventStore/Drivers/AzureHelpers.cs]: Same remark as before: no need for `token` or `do..while`

**Review by VictorNicollet — CHANGES_REQUESTED** [review]
Sorry for the delay in writing this review. I think this is nearly done, just a few minor things.

File: ExampleProject/ExampleProject.csproj (hunk 1 above)
Comment (EslaMx7) [reply ExampleProject/ExampleProject.csproj]: Done!, You may want to bump up the `<PackageReference>` versions after publishing this new version to NuGet.

File: Lokad.AzureEventStore.Cache/AzureCacheProvider.cs (hunk 2 above)
Comment (EslaMx7) [reply Lokad.AzureEventStore.Cache/AzureCacheProvider.cs]: Correct!, Done.

File: Lokad.AzureEventStore/Drivers/AzureHelpers.cs (hunk 5 above)
Comment (EslaMx7) [reply Lokad.AzureEventStore/Drivers/AzureHelpers.cs]: Correct! Done.

File: Lokad.AzureEventStore.Test/Lokad.AzureEventStore.Test.csproj (hunk 4 above)
Comment (EslaMx7) [reply Lokad.AzureEventStore.Test/Lokad.AzureEventStore.Test.csproj]: Done!, You may want to bump up the `<PackageReference>` versions after publishing this new version to NuGet.

File: Lokad.AzureEventStore.Cache/Lokad.AzureEventStore.Cache.csproj (hunk 3 above)
Comment (EslaMx7) [reply Lokad.AzureEventStore.Cache/Lokad.AzureEventStore.Cache.csproj]: Done!, You may want to bump up the `<PackageReference>` versions after publishing this new version to NuGet.

**Review by VictorNicollet — APPROVED** [review]
Looks good, thank you !

**Pull request merged**